config/initializers/mysql2_adapter.rb

```diff
@@ -0,0 +1,3 @@
+class ActiveRecord::ConnectionAdapters::Mysql2Adapter
+  NATIVE_DATABASE_TYPES[:primary_key] = "int(11) auto_increment PRIMARY KEY"
```

Comment:
add mysql2 initializer to create non-null primary keys

Got the error "Mysql2::Error: All parts of a PRIMARY KEY must be NOT NULL; if you need NULL in a key, use UNIQUE instead:"

This was super helpful. https://www.alexfilatov.com/2016/04/06/rails-3-migration-fails-after-mysql-upgrade-to-5-7-10/